### PR TITLE
refactor(app-shell): remove old feature flags

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

I assumed flags are passed from the app but quick-access connects to them itself. Another case for types casue the prop-types are not followed through to the `create-commands`-fn.

### CODE REVIEW

File: packages/application-shell/src/components/quick-access/quick-access.js [hunk 1]
```diff
@@ -356,15 +353,12 @@ export default flowRight(
   injectIntl,
   injectFeatureToggles([
     'pimSearch',
-    'customerGroups',
-    'projectSettings',
-    'developerSettings',
-    'projectSettingsChannels',
```
Comment (tdeekens) [inline packages/application-shell/src/components/quick-access/quick-access.js]: I went ahead and did a little clean up. These feature won't be untoggled and we should clean up our used flags in the MC then too.

**Comment by tdeekens** [comment]
> it looks to me that I have to update this file as well, since I introduced 2 new feature flags

No. Only if those features are ought to be accessible through Quick Access and their access is toggled.

**Comment by adnasa** [comment]
yeah I just realised that now. carry on 😗

File: packages/application-shell/src/components/quick-access/create-commands.js [hunk 2]
```diff
@@ -312,16 +310,15 @@ export default ({
...
-                to: `/${project.key}/settings/developer/api-clients`,
-              },
+          hasPermission(permissions.ManageProject, project.permissions) && {
+            id: 'go/settings/developer',
+            text: intl.formatMessage(messages.openDeveloperSettings),
+            action: {
+              type: 'go',
+              to: `/${project.key}/settings/developer/api-clients`,
             },
-          featureToggles.customApplicationsSettings &&
```
Comment (tdeekens) [inline packages/application-shell/src/components/quick-access/create-commands.js]: This was actually wrong. There is no flag like this and the flag requested is called differently. Hence, quick access for custom app settings was never working to how I see it.

**Pull request merged**